cylc/flow/platforms.py

```diff
@@ -46,12 +46,19 @@ def platform_from_name(platform_name=None, platforms=None):
     """
     if platforms is None:
         platforms = glbl_cfg().get(['platforms'])
+    platform_aliases = glbl_cfg().get(['platform aliases'])
 
     if platform_name is None:
         platform_data = deepcopy(platforms['localhost'])
         platform_data['name'] = 'localhost'
         return platform_data
 
+    for platform_name_re in reversed(list(platform_aliases)):
```

Comment:
Funny thought, what if someone defines both a platform and a platform group with the same name?

* Should we validate platform groups to prevent this from occurring.
* Otherwise the approach used here (go through platform groups before platforms) seems sensible to me.

Reply:
There are cases where this might be reasonable:
```
[platform aliases]
  # Site user prevents anyone specifying a login node (unless they do it explicitly by creating their own platform)
  [[hpc\d?]]
    platforms = hpc1, hpc2   #, hpc3 is out of service, but everyone's hp3 tasks work anyway!
    ...
```
...but equally allows wonderful brainaches...
```
[platform aliases]
  [[chocolate]]
    platforms = vanilla
  [[vanilla]]
    platforms = raspberry
  [[raspberry]]
    platforms = chocolate
```